### ISSUE TITLE
Trust Gate

### ISSUE THREAD

**Issue by TheMingo888** [jasonrohrer/OneLife#964]
The new trust system has already been a huge help in finding nearby allies, but I think it could be expanded to improve other aspects of the game. One main sticking point I often encounter is property gates. When I make a property, I do so to keep out those who would steal or waste the things contained inside. However, I still want to let those I trust access the resources freely.

One way to do this has historically been through the follower gate, which lets anyone that follows you use the gate. The trouble with this is that anyone can follow you at any time, and you have to specifically "exile" people to block their access. In other words, it functions as a "blacklist", letting anyone access the resources in the property by default. For those ne'er-do-wells who seek to steal from you, they can easily overcome the follower gate before you notice, rendering this solution woefully insufficient. Hence, I very rarely see it used.

I'd like to propose a gate less restrictive than the standard property gate, but more restrictive than the follower gate: a "trust gate". Rather than let anyone in that follows you, only those you have trusted can use it. This solves the issue of the follower gate which allows other people to "let themselves in" by following you, while not making the gate so restrictive that only you can use it by default.

Such a gate would have several advantages over giving ownership manually to every person you trust:
- Giving ownership manually requires you to bring each person close to the gate, since you cannot pass ownership to someone who is far away anymore.
- Giving ownership manually requires you to type the full name of each person, which is very tedious if you have several people trusted.
- There may be people you trust that you haven't met in your current life, so you wouldn't have known to give ownership to them. There may also be trusted players born later in your life while you are away from town, preventing them from accessing your property until you return.
- Most importantly, giving ownership to other players allows them to _also_ give ownership to others, creating a chain of ownership transfers that ultimately makes your property very insecure. A trust gate would not allow those you trust to give other people access, since you would (by default) be the only owner (and thus the only person whose trusts matter).

**Comment by cordy-123** [comment]
The problem with that is that items in the property are currently seen (in the current game culture) as a "family's property". 

With your proposal, anyone you trust even if they are from DIFFERENT FAM will be able get into the gate and take anything they want.     
 
---------

Sometimes, non-mod/non-pro players would come up to their family and say, that ginger guy just took a pie or took a bucket of latex, took a horse, etc. 

If it was a zoomout mod player/pro player were to see this, they would know that a traveler taking a bite of pie, or taking 2 peppers and 2 tomatoes (from the brown fam for seeds), taking latex buckets (from brown fam) and dumping empty buckets a few tiles over is just another pro/mod player doing stuff / "meta trading". And they would assess that it's not a "griefer".

But for non-mod/non-pro players, this is something they look at as something to "report" or tell other allies/people in their family as something wrong happening or at the very least, someone to be watched.

---

I hardly think that non-mod/non-pro seeing a diff race / fam person going into the town's property fence and taking 5 tanks of oil and a truck as something that would not cause them alarm. 
(Although of course, the chances of a non-mod player actually seeing it is slim, as most property fences are decently far away from the view of a vanilla player.)

Even for me (a mod player), seeing a random person (which is what I would see) from a different fam coming in with nothing and coming out of the town property with tanks of oil and taking a truck would certainly raise an eyebrow.

**Comment by cordy-123** [comment]
> * Giving ownership manually requires you to type the full name of each person, which is very tedious if you have several people trusted.

Only cursing requires using the full name. 

Giving property ownership does NOT require the full name.

**Comment by cordy-123** [comment]
> * Most importantly, giving ownership to other players allows them to _also_ give ownership to others, creating a chain of ownership transfers that ultimately makes your property very insecure. A trust gate would not allow those you trust to give other people access, since you would (by default) be the only owner (and thus the only person whose trusts matter).
       
What happens during low pop... (or during instances of an account that only trusted very few people)

when there are only a few people and one of the few people left with the "CHAIN trust gate" is from another family.

And no one "trusted" is left in your family.

Example, the trust gate ownership fell to a desert fam member. (You are from the arctic fam.)
You'd have to go 600 plus tiles south for the desert fam expert waystone (and then however many tiles east / west) to get into your OWN fam's property for oil/truck/etc.

**Comment by cordy-123** [comment]
Presumably a "trust gate" would be a different game item from a "property gate"?
 
What happens if the last "trusted" person left has no one ONLINE playing who they trust? 
  
Does the gate just immediately fall?
(When the last person in a family dies, a "property gate" immediately falls. Unless they give the gate to someone else.)

And it's a free for all steal / loot for anyone? 

Bye-bye trucks, engines, oil, etc.

**Comment by TheMingo888** [comment]
> > * Most importantly, giving ownership to other players allows them to _also_ give ownership to others, creating a chain of ownership transfers that ultimately makes your property very insecure. A trust gate would not allow those you trust to give other people access, since you would (by default) be the only owner (and thus the only person whose trusts matter).
> 
> What happens during low pop... (or during instances of an account that only trusted very few people)
> 
> when there are only a few people and one of the few people left with the "CHAIN trust gate" is from another family.
> 
> And no one "trusted" is left in your family.
> 
> Example, the trusted gate fell to a desert fam member. (You are from the arctic fam.) You'd have to go 600 plus tiles south for the desert fam expert waystone (and then however many tiles east / west) to get into your fam's property for oil/truck/etc.



> Presumably a "trust gate" would be a different game item from a "property gate"?
> 
> What happens if the last "trusted" person left has no one ONLINE playing who they trust?
> 
> Does the gate just immediately fall?
> 
> And it's a free for all steal / loot for anyone?

I think you perhaps misunderstood my suggestion. The gate would only have one (1) owner by default. regardless of how many people can actually use it. When that owner dies, the gate passes to someone else in the same family, exactly like a standard property gate. The only difference is that those the owner trusts can also use the gate in addition to the owner, without being owners themselves. If ownership of the gate falls to someone who doesn't make use of the trust system, this isn't a problem at all. It behaves exactly like a regular gate in such a situation.

**Comment by TheMingo888** [comment]
> The problem with that is that items in the property are currently seen (in the current game culture) as a family's property.
> 
> With your proposal, anyone you trust even if they are from DIFFERENT FAM will get the gate.

Yes, this is exactly right. This is also how follower gates work, too, since people from different families can all follow the same leader. The advantage of this "trust gate" is that the owner is in full control of who can and cannot use the gate, unlike with the follower gate where people can gain access without the owner's consent.

And I think it's fine if people from other families can use the gate, assuming I have them trusted. After all, I don't trust thieves and griefers. Which, by the way, is the reason property fences and gates are useful in the first place—to keep those that would steal away. (If someone I do have trusted does steal, that would give me a great excuse to promptly un-trust them too.)

Remember, the current solution to letting those from other families use a property gate is worse. If you give ownership to someone from another family, and they end up being the last person alive who owns the gate, then when they die, the gate falls to _their_ family. If this is the communal town property, you end up having to use an elder removal note to punch a hole in the fence so your family can regain access, which is really annoying. With this proposal, there is no need to make them owner in the first place.

Also... this isn't only useful for the communal town property. Those can stay using standard property gates if the builders so choose. Having more options for protecting projects outside of town is also a key desire.

---

It would be nice if it was easy to switch between a standard property gate and a trust gate. Currently, it is rather annoying to switch from a follower gate back to a standard gate, since you have to knock it down and rebuild it, and the skewer is lost. Maybe that could be improved as well (maybe by letting the owner cut off the skewer or whatnot with a flint chip or knife)?

**Comment by DiscardedSlinky** [comment]
It's a cool idea 👍

**Comment by cordy-123** [comment]
> (If someone I do have trusted does steal, that would give me a great excuse to promptly un-trust them too.)

How would you know if someone you trusted "steals" something?

Since there is no communication needed, the trusted person will just go in the gate and take stuff (all within a couple of secs or at most around 2 minutes).

Also, that presupposes that you saw the trusted person take the items in those 2 minutes. 

----

The reason for the property gate is for **people to interact / talk and make the decision** (to the best of your ability) on what resources you want to give out / trade out. 
(How many tanks of oil do you think 5 buckets of latex is worth?
There's only 3 tanks of oil left. Should you use it for water or for iron? How many can you spare? etc, etc.) 

**The human element to the game.** 

To make stories and for the associated gameplay entailed in such roles.

Your role as the property owner is being the town's property gatekeeper (and honestly, usually also main trade person for your town, for high value items at least).

Someone has chosen you (thru gene score / someone else giving you ownership) to be the steward of the town's most valuable items.

----

Quite a bit of interaction is already lost with the "meta" ways to do things.

Want bananas for craving? Good chance that if you scan the jungle area near a semi-developed jungle town that someone already placed bananas in the mountain biome (so any race can access them).

Need sulfur or latex? Just bring empty bowls or buckets. Good chance there's some lying around the expert fam. Just take the bowls of sulfur and drop the empty bowls. Trade with no interaction needed. 
(These are low value items compared to trucks and oil tanks and as such are usually left lying around town.)

**Comment by cordy-123** [comment]
I still think letting "trusted" players from **OTHER fams** go in and get anything they want from your town's property that you _by happenstance inherited through MEME score_ (etc.) is **CRAZY**.      
     
But if Mr. Rohrer does agree with you and changes it.   

I'd rather have it be an **opt-in system**.

**Rather than every gate I own can automatically be opened by every player I ever trusted.** 
(Which is in contradiction to Mr. Rohrer original idea of private property (for example family house, family farm) passed to children / others and his subsequent idea of property gates as safekeeping for items such as oil, water / wells, etc.. (Yes, Mr. Rohrer once suggested property fencing the well and for someone to take up the role of well property owner ie. the "well master".)
(This presupposes that Mr. Rohrer combines the trust thing and the normal property gate. Not that he makes a new "trust gate" game item and sprite.)

Some sort of phrase. 

(Example, currently you can do My children own this, My family owns this.)

Some sort of phrase that allows "trusted" players to **open** (BUT NOT OWN or destroy) the gate.

[Credit for the phrase idea to Mingo: ""those i trust own this"
"https://discord.com/channels/328215300279500800/328215300279500800/1163938571653226650] (but NOT own)

**Comment by TheMingo888** [comment]
> I still think letting "trusted" players from **OTHER fams** go in and get anything they want from your town's property that you _by happenstance inherited through MEME score_ (etc.) is **CRAZY**.
> 
> But if Mr. Rohrer does agree with you and changes it.
> 
> I'd rather have it be an **opt-in system**.
> 
> **Rather than every gate I own can automatically be opened by every player I ever trusted.** (Which is in contradiction to Mr. Rohrer original idea of private property (for example family house, family farm) passed to children / others and his subsequent idea of property gates as safekeeping for items such as oil, water / wells, etc.. (Yes, Mr. Rohrer once suggested property fencing the well and for someone to take up the role of well property owner ie. the "well master".) (This presupposes that Mr. Rohrer combines the trust thing and the normal property gate. Not that he makes a new "trust gate" game item and sprite.)
> 
> Some sort of phrase.
> 
> (Example, currently you can do My children own this, My family owns this.)
> 
> Some sort of phrase that allows "trusted" players to **open** (BUT NOT OWN or destroy) the gate.
> 
> [Credit for the phrase idea to Mingo: https://discord.com/channels/328215300279500800/328215300279500800/1163938571653226650]

My proposal is not to change the behaviour of all gates. It is to add a new gate. Thus, it would inherently be an opt-in system.

After thinking about it, I rejected the phrase idea for a few reasons:
- I don’t want to grant ownership to others. I only want to allow them to use the property and not transfer it further.
- If someone from another family got ownership, and they are the last owner alive, it causes issues where the other family inherits the gate instead of my family. (You pointed this out already above.)
- Such a phrase would only grant ownership to those within 200 tiles of the gate at the moment I say it, rather than all current and future trusted players.

The idea is based off the model of the follower gate, where you can “modify” its behaviour by adding a skewer. I suggest something similar with a different modifying item. It would be nice if owners could “un-modify” both gates and retrieve the spent item, too.

**Comment by Tarr-OHOL** [comment]
I don’t think there’s an issue with a trust gate overall. It is however a weird meta version of a gate as it would only be passed through to people you’ve played with previously. 

Current gates only exist as private to private (passed from person to person as long as they continue to chain back to each other.) 

Private to public (Gate will get passed to leader once you pass if you do not specifically to a person or specific persons) 

Pubic/Follower gates (Gates that are passed down by the owner to everyone or are denoted for use by follower.)

A trust based gate would allow vanilla players to have a white list gate system that allows players they have previously had good experiences to get into resources. My only concern is I don’t exactly know how many players have used /trust meaning these gates may just break instantly once the own dies or will end up defaulting to the town leader.

**Comment by SoloAceMouse** [comment]
Just got caught up reading this, and I think most of the criticism levelled doesn't really apply to the idea as Mingo presented it, frankly.

-----

**I support and approve of the addition of trust gates**

As a vanilla [non-modded] player, this addition would give me the ability to white-list others for valuable resource/item access. This would help bridge the gap between the modded players and vanilla in giving more options in selecting who has access, which modded players currently have an obvious advantage in.

Additonally, anyone who does not see the utility in a trust gate is already served by the existence of standard prop gates.

-----

Short version:
**TRUST GATE IS GOOD, PLEASE ADD TO GAME**

**Comment by jasonrohrer** [comment]
I will read more of this thread, but my initial thought is this:

There has to be a careful balance between "living a unique, self-contained life" (the whole point of the game) and these global, persistent systems that are necessary to deal with serial griefers.  Curses are one example.  The trust system is another example.  Also global systems to help motivating congruent player behaviors.  Fitness scores are an example of that.

I hate these systems.  They are "magic" and fly in the face of what this game is about (the "veil of ignorance," etc).  "You" don't exist in this game.  But these systems exist as band-aides for other problems that I failed to solve inside a given life.

Ideally, criminals would hesitate to be criminals because they have too much to lose, but the reality is that "getting born and growing up again" is just too low of a cost.  So other costs were added.

Ideally, people would be highly motivated to take care of their children and ensure their survival.  But "meta goals" take over (goals that players have across lives, like finally building an engine or whatever).  The original idea is that there'd be no point in building something if there are no survivors to use it.  But that wasn't enough of a benefit.  Fitness scores were one attempt at fixing this, but even that wasn't enough.  Eventually, Legacy Chains fixed this (see:  http://onehouronelife.com/newsPage.php?postID=10324 ).

Anyway, as much as possible, you're supposed to be starting from scratch with each new life.  There's not supposed to be a big club that forms over time, across lives.

The Trust system violates this goal, of course.  But it was just so damn hard to deal with griefers in this environment.  All the things that make each life truly fresh also help griefers disguise themselves.

Still, there is supposed to be some "figuring stuff out" in each life.  Some messiness.  Figuring out who to add to the gate, etc.

I don't want you to be able to figure this out, once and for all, globally across lives and time.  "Ah... now that I've been playing for a while, my gate list is perfectly ironed out forever."

**Comment by jasonrohrer** [comment]
And yeah, it is definitely weird to imagine some "stranger" to the family just walking in and magically opening the gate without any other interaction.  Who is this guy?  None of us (not even the gate's owner) have ever seen this guy before, nor spoken to him.  We don't even know his name.  But somehow, he can just walk in here from far away and open this gate?

This flies to heavily in the face of the granular nature of lives in the game.

The current Trust System is implemented with a very light touch.  It affects nothing, other than a little "+" that appears in their speech.  Just a little hint for you to see, nothing more.

There are still weird things that can come of this, but they are meta effects.  Like some stranger shows up to the village, and your mother is instantly friendly to her.  "Mom, why did you just let that stranger in and let them borrow our delivery truck?"  "Don't worry, sweetie, I trust them."  That is at least an interesting story that's congruent with the themes of the game.  In real life, sometimes you just get a good feeling about someone.  The "+" you see in their speech is that good feeling.

And note that it doesn't tell you *that* much about them.  It doesn't tell you who they actually are, for example.  You can't know that this is Tarr coming along.
